xayn-ai/src/coi/point.rs

```diff
@@ -6,6 +6,50 @@ use serde::{Deserialize, Serialize};
 
 use crate::{coi::CoiId, embedding::utils::Embedding, utils::system_time_now};
 
+#[derive(Clone, Copy, Deserialize, Serialize)]
+#[cfg_attr(test, derive(Debug))]
+pub(crate) struct CoiView {
+    pub(crate) count: usize,
+    pub(crate) time: Duration,
+    pub(crate) last: SystemTime,
+}
+
+impl CoiView {
+    pub(crate) fn new(viewed: Option<Duration>) -> Self {
+        Self {
+            count: 1,
+            time: viewed.unwrap_or_default(),
+            last: system_time_now(),
```

Comment:
for now this is testable pretty well. for coi comparisons the stats don't matter, they are excluded from `PartialEq`, because the usage stats are somewhat meta data to the defining properties of a coi (like the embedding).

for tests which use the stats, like the tests for `compute_weights()`, there is a small time difference between `create_pos_cois()` and `compute_weights()` which can be accounted for by the `epsilon`.

if we were to introduce a closure in the constructors, then we would have to do the same for `compute_weights()` (only one of them wouldn't help much). this brings additional complexity which can be avoided, as long as we don't need it, since we can currently write test without it.

Reply:
Yes if we use a closure we need to change it everywhere. It is similar to the trait `BetaSampler`. If it is not creating problem for testing we can leave it like this and consider to change it later if needed.